### PR TITLE
lint: fix lsp warning in `vim.lsp.inlay_hint.is_enabled`

### PR DESCRIPTION
**Pull request by srdtrk**
***************************************************************************
**NOTE**
Please verify that the `base repository` above has the intended destination!
Github by default opens Pull Requests against the parent of a forked repository.
If this is your personal fork and you didn't intend to open a PR for contribution
to the original project then adjust the `base repository` accordingly.
**************************************************************************

### CODE REVIEW

File: init.lua [hunk 1]
```diff
@@ -542,7 +542,7 @@ require('lazy').setup({
           -- This may be unwanted, since they displace some of your code
           if client and client.server_capabilities.inlayHintProvider and vim.lsp.inlay_hint then
             map('<leader>th', function()
-              vim.lsp.inlay_hint.enable(not vim.lsp.inlay_hint.is_enabled())
+              vim.lsp.inlay_hint.enable(not vim.lsp.inlay_hint.is_enabled {})
```
Comment (tjkuson) [inline init.lua:545]:
```suggestion
              vim.lsp.inlay_hint.enable(not vim.lsp.inlay_hint.is_enabled { bufnr = event.buf })
```

I think this is needed as well.

Comment (srdtrk) [reply init.lua]: done

**Comment by feoh** [comment]
Hi! Can you please help me understand what this change is meant to fix? Where are you seeing this warning?

Thanks for your contribution!

**Comment by VlaDexa** [comment]
This was a [neovim issue](https://github.com/neovim/neovim/issues/29230) with their documentation where they showed that `vim.lsp.inlay_hint.is_enabled` requires an argument, when it actually doesn't.

Also [someone reported](https://github.com/nvim-lua/kickstart.nvim/pull/936#discussion_r1608524444) that omitting bufnr causes *issues*, so I included it in #936

**Comment by feoh** [comment]
Thanks so much @VlaDexa - merging!

**Pull request merged**